From: Borislav Petkov <bp@suse.de>
Date: Sat, 16 Dec 2017 12:57:44 +0100
Subject: x86/MSR: Move native_*msr(.. u64) to msr.h
Patch-mainline: submitted on 2018/1/9
References: bsc#1068032

Move them to the msr.h header for a wider use.

No functionality change.

Well, we don't have them in 3.x so add them.

Signed-off-by: Borislav Petkov <bp@suse.de>
---
 arch/x86/include/asm/msr.h |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

--- a/arch/x86/include/asm/msr.h
+++ b/arch/x86/include/asm/msr.h
@@ -135,6 +135,21 @@ static inline unsigned long long native_
 	return EAX_EDX_VAL(val, low, high);
 }
 
+#define native_rdmsr(msr, val1, val2)			\
+do {							\
+	u64 __val = native_read_msr((msr));		\
+	(void)((val1) = (u32)__val);			\
+	(void)((val2) = (u32)(__val >> 32));		\
+} while (0)
+
+#define native_wrmsr(msr, low, high)			\
+	native_write_msr(msr, low, high)
+
+#define native_wrmsrl(msr, val)				\
+	native_write_msr((msr),				\
+			 (u32)((u64)(val)),		\
+			 (u32)((u64)(val) >> 32))
+
 #ifdef CONFIG_PARAVIRT
 #include <asm/paravirt.h>
 #else
